tools/buildsys/src/builder.rs

```diff
@@ -255,6 +281,106 @@ where
         .collect()
 }
 
+/// Create a directory for build artifacts.
+fn create_build_dir(kind: &BuildType, name: &str) -> Result<PathBuf> {
+    let prefix = match kind {
+        BuildType::Package => "packages",
+        BuildType::Variant => "variants",
+    };
+
+    let path = [&getenv("BUILDSYS_STATE_DIR")?, prefix, name]
```

Comment:
At some point it might be nice to move all of these variables to a StructOpt, which can get its values from environment variables. This would make it easier to understand all of the inputs to the program by centralizing them.